File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        count = 0
        size = len(nums)
        index = 0
        idx = 1
        
        while(idx<len(nums)):
            flag = 0
            while idx<len(nums) and nums[idx] == nums[idx-1]:
                flag = 1
                del nums[idx]
                
            if flag == 0:
                idx+=1
        
        
        
        
        
        """while(index<len(nums)):         
                if(index+1 <len(nums) and nums[index] == nums[index+1]):
                    idt = index
                    while(idt+1 <len(nums) and nums[idt] == nums[idt+1]):
                        del nums[idt]

                index+=1
                idt=0"""
        return len(nums)

File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int firstDuplicate = 0, size = nums.length, totalDuplicates = 0;
        int ptr = 0, ptr2 = 0;
        while(ptr < size) {
            int temp = nums[ptr];
            while( ptr<size && (ptr-1) >= 0 && nums[ptr-1] == nums[ptr]) {
                ptr++;
            }
            if(ptr >= size) {
                break;
            }
            //System.out.println("s " + ptr2 + " " + ptr + " -- " + nums[ptr2] + " " + nums[ptr]);
            nums[ptr2] = nums[ptr];
            ptr++;
            ptr2++;
        }
        //System.out.println("res ::" + ptr2);
        return ptr2;
    }
}